crates/protoc-gen-grafbase-subgraph/src/display_utils.rs

```diff
@@ -0,0 +1,35 @@
+use std::fmt;
+
+pub(crate) fn display_fn(f: impl Fn(&mut fmt::Formatter<'_>) -> fmt::Result) -> impl fmt::Display {
+    struct DisplayFn<F>(F);
+
+    impl<F> fmt::Display for DisplayFn<F>
+    where
+        F: Fn(&mut fmt::Formatter<'_>) -> fmt::Result,
+    {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            (self.0)(f)
+        }
+    }
+
+    DisplayFn(f)
+}
+
+pub(crate) fn grpc_path_to_graphql_name(path: &str) -> impl fmt::Display {
+    display_fn(|f| {
```

Comment:
Seems like you're re-inventing: https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.format_with

Reply:
true